TACC/protx-dashboard
# task/COOKS-375: rework changing between maltreatment/analytics/demographics

## Overview: ##

This PR reworks the user changing between maltreatment/analytics/demographics:
* Added element for the user to change between maltreatment/analytics/demographics
* added redux state and related sagas for current selection (i.e. year, maltreatment, etc) - [COOKS-55](https://jira.tacc.utexas.edu/browse/COOKS-55)
* fixed bug we had seen when switching between percentages and totals for a demographic feature that shouldn't be displayed. (in the bug you would see a blank map; this can be seen on prod)

## Related Jira tickets: ##

* [COOKS-375](https://jira.tacc.utexas.edu/browse/COOKS-375)
* [COOKS-55](https://jira.tacc.utexas.edu/browse/COOKS-55)


## Testing Steps: ##
1. Go to [ /protx/dash/](https://cep.test/protx/dash/)
2. Switch between maltreatment, analytics, demographics
3. click on different counties and switch between reports to confirm your selection, e.g year, county, (if possible) stays between reports. 

Review README
1) check if README makes sense

## Notes: ##

- [x] clarify UI element to switch between analytics/demographics/maltreatment .  See John's mockup.  **update: keep as currently in  pr
- [x] clarify route (i.e. dropping routes now but easy to add with q**uery params that dispatch saga actions)
- [x] add unit tests for configuration.js reducer. **Update: added follow on isssue: https://jira.tacc.utexas.edu/browse/COOKS-385)**
- [x] clarify if we want a separate state for unit (percent/count etc) for demographics vs maltreatment (we are using the same state for this at the moment). same applies for year  **update: keep as currently in  pr**
